test/Microsoft.AspNetCore.Mvc.Core.Test/Formatters/MediaTypeTest.cs

```diff
@@ -171,6 +205,17 @@ public void GetParameter_IsCaseInsensitive()
         [InlineData("application/json", "application/json;format=indent;charset=utf-8")]
         [InlineData("application/json;format=indent;charset=utf-8", "application/json;format=indent;charset=utf-8")]
         [InlineData("application/json;charset=utf-8;format=indent", "application/json;format=indent;charset=utf-8")]
+        [InlineData("application/*", "application/json")]
+        [InlineData("application/*", "application/entitytype+json;v=2")]
+        [InlineData("application/*;v=2", "application/entitytype+json;v=2")]
+        [InlineData("application/json;*", "application/json;v=2")]
+        [InlineData("application/json;v=2;*", "application/json;v=2;charset=utf-8")]
+        [InlineData("*/*", "application/json")]
+        [InlineData("application/entity+json", "application/entity+json")]
+        [InlineData("application/*+json", "application/entity+json")]
+        [InlineData("application/entity+*", "application/entity+json")]
+        [InlineData("application/*+*", "application/entity+json")]
```

Comment:
They are odd cases, but I don't see any reason to prohibit them - they fit into existing patterns and nothing bad happens with them.

Someone might have 5 different syntactical representations of their entity (`application/myentity+json`, `...+xml`, `...+txt`, ...) and want to have a single `myentity` formatter that looks at the media type suffix and deals with all of them for that entity type. I know that would be weird (vs having formatters that deal with syntaxes rather than entity types) but I don't see why we'd want to prohibit it.